### PR TITLE
Separatrix mods for up/down split

### PR DESCRIPTION
**Pull request by geograham**
## Description

Current code is written assuming in/out split for the double null separatrix list found using find_LCFS_separatrix. 
Relevant Functions have been modified to allow for an up/down split. 

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/equilibria/find.py [hunk 1]
```diff
@@ -832,34 +873,125 @@ def get_legs(
...
-    separatrix = equilibrium.get_separatrix()
-    delta = equilibrium.grid.dx
+    delta = np.max(equilibrium.grid.x) - np.min(equilibrium.grid.x)
+    delta_offsets = equilibrium.grid.dx
     dx_offsets = None if n_layers == 1 else np.linspace(0, dx_off, n_layers)[1:]
 
     if isinstance(separatrix, list):
-        # Double null (sort in/out bottom/top)
-        separatrix.sort(key=lambda half_sep: np.min(half_sep.x))
+        # Double null (N.B., always sort resulting leg dict by in/out
```
[the remark this answers is not in the document]
Comment (geograham) [reply bluemira/equilibria/find.py:893]: Ok, no problem, will split it up and try to make it clearer.

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]
Really quick review from my side... apologies have not had time to look at the actual algorithm or play around with it. Just a couple of minor first impressions

File: bluemira/equilibria/find.py [hunk 2]
```diff
@@ -650,6 +647,8 @@ def find_LCFS_separatrix(
     x_points: Optional[List[Xpoint]] = None,
     double_null: bool = False,
     psi_n_tol: float = 1e-6,
+    delta_start: float = 0.01,
+    rtol: float = 1e-2,
```
[the remark this answers is not in the document]
Comment (geograham) [reply bluemira/equilibria/find.py]: thanks for spotting, have changed it back

**Review by CoronelBuendia — APPROVED** [review]

**Review by je-cook — COMMENTED** [review]
Looks good just one minor

File: bluemira/equilibria/find_legs.py [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+            # Check to determine configuration (separatrix list is sorted by
+            # loop length when it is found (longest first), so use [0])
+            z_sep = min(abs(min(self.separatrix[0].z)), abs(max(self.separatrix[0].z)))
+            legs_upper_lower = np.isclose(
+                z_sep, self.abs_z_lcfs[0], rtol=self.rtol
+            ) or np.isclose(z_sep, self.abs_z_lcfs[1], rtol=self.rtol)
+            if legs_upper_lower:
+                # Sort LOWER then UPPER when use get_legs
+                # self.separatrix remains sorted by loop length
+                return NumNull["DN".upper()], SortSplit["Z".upper()]
```
Comment (je-cook) [inline bluemira/equilibria/find_legs.py:125]:
```suggestion
                return NumNull.DN, SortSplit.Z
```

File: bluemira/equilibria/find_legs.py [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+            # Sort IN then OUT when use get_legs
+            # self.separatrix remains sorted by loop length
+            return NumNull["DN".upper()], SortSplit["X".upper()]
```
Comment (je-cook) [inline bluemira/equilibria/find_legs.py:128]:
```suggestion
            return NumNull.DN, SortSplit.X
```

File: bluemira/equilibria/find_legs.py [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+        # --- Single Null ---
+        self.x_points = self.x_points[0]
+        return NumNull["SN".upper()], SortSplit["X".upper()]
```
Comment (je-cook) [inline bluemira/equilibria/find_legs.py:131]:
```suggestion
        return NumNull.SN, SortSplit.X
```

**Review by je-cook — APPROVED** [review]
Couple of quick fixes otherwise good to go!

**Comment by je-cook** [comment]
I don't know what up with the CI...everything passes locally

**Pull request merged**